### ISSUE TITLE
Error using Python 3.8 on MacOS 10.14

### ISSUE THREAD

**Issue by jimmywags** [trailofbits/algo#1622]
Working on MacOS. Installed Python. Installed dependencies. Added users to config. No other edits to config.

Ran ./algo from algo directory. Received the following message:
```
[WARNING]: Could not match supplied host pattern, ignoring: vpn-host


PLAY [localhost] ***************************************************************
ERROR! Unexpected Exception, this is probably a bug: cannot pickle '_io.TextIOWrapper' object
```

**Comment by davidemyers** [comment]
You can ignore the `WARNING`, the problem is the `ERROR`.

What version of macOS? Which Python 3 did you choose to install? Could there be more than one installation of Python 3 on your system?

**Comment by jimmywags** [comment]
MacOS 10.14.6
Python 3.8.0

And it seems I also have Python 2.7.10 installed. Didn't realize that. I assume the second version of Python is interfering. Sorry. I'm a command line novice. Should I remove Python 2?

**Comment by davidemyers** [comment]
macOS Mojave comes with Python 2.7.10 installed as `python` and there's no need to remove it. Just ignore it and run `python3` as instructed in the Algo README. I was concerned maybe you had two versions of Python 3 installed, which is possible depending on where you get it from.

Python 3.8 was only just released and I wonder if that's causing the problem. After some quick searching I haven't found anyone complaining about problems using Ansible with Python 3.8 so I'm not sure what's going on.

**Comment by davidemyers** [comment]
I can reproduce the error using Python 3.8 from `python.org` on macOS 10.14.6.

Please change the title of this issue something like: "Error using Python 3.8 on macOS".

**Comment by davidemyers** [comment]
Python 3.7.5 appears to work so you can use that as a workaround for now.

**Comment by jimmywags** [comment]
Thank you David. I will retry in the morning.

**Comment by davidemyers** [comment]
As far as I can determine (and I may very well be wrong about this), Python made a change in 3.8 to how subprocesses are created on macOS, and this has caught out Ansible 2.8. The issue should be fixed in a future version of Ansible.

I'm not sure there's much Algo can do except note in the README that macOS users must use Python 3.7.X.

**Comment by jimmywags** [comment]
Success. I removed Python 3.8, installed Python 3.7.5 and setup went smoothly. I appreciate all your prompt feedback and work on this project.

**Comment by TC1977** [comment]
@davidemyers should I add a quick note about this into #1607, or make a separate PR?

**Comment by davidemyers** [comment]
Put it in #1607. I suggest:

```patch
diff --git a/README.md b/README.md
index d13bf61..5d775c3 100644
--- a/README.md
+++ b/README.md
@@ -37,12 +37,12 @@ The easiest way to get an Algo server running is to run it on your local system
 
     - Run the command `git clone https://github.com/trailofbits/algo.git` to create a directory named `algo` containing the Algo scripts.
 
-3. **Install Algo's core dependencies.** Algo requires that **Python 3** and at least one supporting package are installed on your system.
+3. **Install Algo's core dependencies.** Algo requires that **Python 3.6 or later** and at least one supporting package are installed on your system.
 
     - **macOS:** Apple does not provide Python 3 with macOS. There are two ways to obtain it:
         * Use the [Homebrew](https://brew.sh) package manager. After installing Homebrew install Python 3 by running `brew install python3`.
 
-        * Download and install the latest stable [Python 3 package](https://www.python.org/downloads/mac-osx/). Be sure to run the included *Install Certificates* command from Finder.
+        * Download and install the latest stable [Python 3.7 package](https://www.python.org/downloads/mac-osx/) (currently Python 3.8 will not work). Be sure to run the included *Install Certificates* command from Finder.
 
         Once Python 3 is installed on your Mac, from Terminal run:
         ```bash
```
This may need to be adjusted once Homebrew upgrades to 3.8.

I think Ansible is aware of the issue and there's already a fix in the version under development.

**Comment by TC1977** [comment]
Ok, done.

**Comment by davidemyers** [comment]
@jackivanov Python 3.8.1 from python.org now works on macOS when using Ansible 2.9.2 (the fix was in 2.9.1). I tested a deploy to DigitalOcean from Mojave.

Ansible 2.9.2 also works deploying to DO from Ubuntu 19.10 with the default Python 3.7.5.

**Comment by disconnect3d** [comment]
This occured for me on (the default) Python 3.8.6 on MacOS.

Maybe we could check either for Python version, or, the ansible output in the `./algo` script and inform about the solution (here: installed Py 3.9 via homebrew and created another venv from it).

A bit off topic, the `./algo --help` could have a custom info instead of passing the `--help` to `ansible-playbook`.
